### PR TITLE
Added Prims Algorithm under graphs

### LINKED ISSUE #106: [Feature Request] Add prims Algorithm under graph
### Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is. Example: "I'm frustrated when..."

### Additional context

---

**Would you like to work on this feature?**  
- [ ] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by shravya312**
### Description
Added introduction, working, step by step Explaination, Complexity analysis, application, Comparison with Kruskal's Algorithm for Prims Algorithm.

Fixes #106 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/graphs/Prims_Algorithm.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Let’s illustrate Prim's Algorithm with a simple example:
+
+- **Graph Representation:**
+   (B)
+  / | \
+2/  |4  \3
+(A)--(C)
+  \  | \
+  1\ |  \5
+    (D)
```
Comment (ajay-dhangar) [inline docs/graphs/Prims_Algorithm.md]: replace 

```md
   (B)
  / | \
2/  |4  \3
(A)--(C)
  \  | \
  1\ |  \5
    (D)
```

with

````md
```
   (B)
  / | \
2/  |4  \3
(A)--(C)
  \  | \
  1\ |  \5
    (D)
```
````

also, you can use mermaid graph

````md
```mermaid
graph TB
    B -->|2| A
    B -->|4| C
    B -->|3| D
    A ---|1| D
    A --- C
    C -->|5| D
```
````

File: docs/graphs/Prims_Algorithm.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+## Table of Contents
+- [Introduction](#introduction)
+- [How Prim's Algorithm Works](#how-prims-algorithm-works)
+- [Step-by-Step Execution](#step-by-step-execution)
+- [Time Complexity](#time-complexity)
+- [Applications](#applications)
+- [Comparison with Kruskal's Algorithm](#comparison-with-kruskals-algorithm)
+- [Conclusion](#conclusion)
+
```
Comment (ajay-dhangar) [inline docs/graphs/Prims_Algorithm.md]: delete the table of contents, we have default setup for this in sidebar

File: docs/graphs/Prims_Algorithm.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Prim's Algorithm for Minimum Spanning Tree
```
Comment (ajay-dhangar) [inline docs/graphs/Prims_Algorithm.md]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

File: docs/graphs/PrimsAlgorithm.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Prim's Algorithm for Minimum Spanning Tree
```
Comment (ajay-dhangar) [inline docs/graphs/PrimsAlgorithm.md]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions and also, delete `PrimsAlgorithm.md` file and replace your file name `Prims_Algorithm.md` to `prims-algorithm.md` for best practices

**Comment by shravya312** [comment]
Sure I will make changes according what you have suggested and send pr again

File: docs/graphs/Prims_Algorithm.md (hunk 1 above)
Comment (shravya312) [reply docs/graphs/Prims_Algorithm.md]: > replace
> 
> ```
>    (B)
>   / | \
> 2/  |4  \3
> (A)--(C)
>   \  | \
>   1\ |  \5
>     (D)
> ```
> 
> with
> 
> ```
> ```
>    (B)
>   / | \
> 2/  |4  \3
> (A)--(C)
>   \  | \
>   1\ |  \5
>     (D)
> ```
> ```
> 
> also, you can use mermaid graph
> 
> ```
> ```mermaid
> graph TB
>     B -->|2| A
>     B -->|4| C
>     B -->|3| D
>     A ---|1| D
>     A --- C
>     C -->|5| D
> ```
> ```

Sure I will make the changes to graph and I will rename my file name as prims-algorithm.md

File: docs/graphs/Prims_Algorithm.md (hunk 2 above)
Comment (shravya312) [reply docs/graphs/Prims_Algorithm.md]: Okay I will do in the format which u have quoted I will delete this part

**Comment by shravya312** [comment]
I am making changes once done I will add new commit to this pr

**Comment by shravya312** [comment]
I have made all the changes suggested Can u review it @ajay-dhangar 
image
image
image
image
image
image

File: docs/graphs/Prims-Algorithm.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
+---
+id: Prims Algorithm
```
Comment (ajay-dhangar) [inline docs/graphs/Prims-Algorithm.md]: replace `id: Prims Algorithm` with `id: prims-algorithm`

File: docs/graphs/Prims-Algorithm.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+title:  Prims Algorithm
+sidebar_label:  prims-algorithm
```
Comment (ajay-dhangar) [inline docs/graphs/Prims-Algorithm.md]: replace `prims-algorithm` with `Prims Algorithm`

File: docs/graphs/Prims-Algorithm.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+sidebar_position: 2
+description: Prim's algorithm is a greedy algorithm that finds the Minimum Spanning Tree (MST)
```
Comment (ajay-dhangar) [inline docs/graphs/Prims-Algorithm.md]: replace `description: Prim's algorithm is a greedy algorithm that finds the Minimum Spanning Tree (MST)` with `description: "Prim's algorithm is a greedy algorithm that finds the Minimum Spanning Tree (MST)"`

File: docs/graphs/Prims-Algorithm.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+tags: [Graph Theory, Prim's Algorithm, Minimum Spanning Tree, Greedy Algorithms, Optimization]
```
Comment (ajay-dhangar) [inline docs/graphs/Prims-Algorithm.md]: replace `Prim's Algorithm` with `prim-algorithm` in tags

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code for best practices

**Comment by shravya312** [comment]
can u review the code made changes suggested @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by shravya312** [comment]
Thanks for guiding me @ajay-dhangar 😇

**Comment by shravya312** [comment]
Thanks for the guidance @ajay-dhangar

**Pull request merged**